doc/docs/Python_Tutorials/Mode_Decomposition.md

```diff
@@ -306,6 +306,138 @@ In the limit where the grating periodicity is much larger than the wavelength an
 
 To convert the diffraction efficiency into transmittance in the *x* direction (in order to be able to compare the scalar-theory results with those from Meep), the diffraction efficiency must be multiplied by the Fresnel transmittance from air to glass and by the cosine of the diffraction angle. We compare the analytic and simulated results at a wavelength of 0.5 μm for diffraction orders 1, 3, 5, and 7. The analytic results are 0.3886, 0.0427, 0.0151, and 0.0074. The Meep results are 0.3891, 0.04287, 0.0152, and 0.0076. This corresponds to relative errors of approximately 1.3%, 0.4%, 0.8%, and 2.1% which indicates good agreement.
 
+We can also use the complex mode coefficients to compute the phase of the diffraction orders. This can be used to generate a phase map of the binary grating as a function of its geometric parameters. Phase maps are important for the design of subwavelength phase shifters such as those used in metalenses. In this demonstration, which is adapted from the previous example, we compute the transmittance spectra and phase map of the zeroth diffraction order (at 0°) for an E<sub>z</sub>-polarized planewave pulse spanning wavelengths of 0.4 to 0.6 μm which is normally incident on a binary grating with a periodicity of 0.35 μm and height of 0.6 μm. The duty cycle of the grating is varied from 0.1 to 0.9 in separate runs.
+
```

Comment:
paragraph to add:

Note that it is generally only the *relative* phase (the phase difference) between different structures.  The overall mode coefficient α is multiplied by a complex number given by the source amplitude, as well as an arbitrary (but deterministic) phase choice by our mode solver (MPB)—but as long as you keep the source fixed as you vary the parameters of the structure, the relative phases are meaningful.  More generally, for metasurface design it is preferable to use the overall complex amplitude α (both phase and amplitude), as we describe in [this 2018 paper](https://www.osapublishing.org/oe/abstract.cfm?uri=oe-26-26-33732): you can use α to optimize for a desired wavefront, as in equation (3) in section 3.1, or more generally for some arbitrary function of the far field as in section 3.2.

Reply:
added.